lib/spack/spack/install_test.py

```diff
@@ -3,34 +3,76 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 import base64
+import contextlib
+import enum
 import hashlib
+import inspect
+import io
 import os
 import re
 import shutil
+import sys
+from collections import Counter, OrderedDict
+from typing import Callable, List, Optional, Tuple, Type, TypeVar, Union
 
 import llnl.util.filesystem as fs
 import llnl.util.tty as tty
+from llnl.util.lang import nullcontext
+from llnl.util.tty.color import colorize
 
 import spack.error
 import spack.paths
 import spack.util.spack_json as sjson
 from spack.spec import Spec
 from spack.util.prefix import Prefix
+from spack.util.string import plural
 
+#: Stand-alone test failure info type
+TestFailureType = Tuple[BaseException, str]
+
+#: Name of the test suite's (JSON) lock file
 test_suite_filename = "test_suite.lock"
+
+#: Name of the test suite results (summary) file
 results_filename = "results.txt"
 
+#: Name of the Spack install phase-time test log file
+spack_install_test_log = "install-time-test-log.txt"
+
+
+ListOrStringType = Union[str, List[str]]
+LogType = Union["tty.log.nixlog", "tty.log.winlog"]
+
+Pb = TypeVar("Pb", bound="spack.package_base.PackageBase")
+TestPackageType = Union[Pb, Type[Pb]]
```

Comment:
(For a follow up PR) I would call this:
```suggestion
PackageObjectOrPackageClass = Union[Pb, Type[Pb]]
```

Reply:
How about `PackageObjectOrClass`?

Reply:
It's a little tighter and IMO conveys the same types.

Reply:
Changed to the name I mentioned.